src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -673,3 +675,324 @@ The Run handle/object supports few other use-cases.
+           xrtAIEResetArray(device_handle)
+           
+           xrtGraphHandle graph = xrtGraphOpen(device, xclbin_uuid, "graph_name");
+           // Graph Reset
+           xrtGraphReset(graphHandle);
+
+
+
+
+Graph execution
+~~~~~~~~~~~~~~~
+
+XRT provides basic graph execution control APIs to initialize, run, wait, and terminate graphs for a specific number of iterations. Below we will review some of the common graph execution styles. 
+
+Graph execution for a fixed number of iterations
+************************************************
+
+A graph can be executed for a fixed number of iterations followed by a "busy-wait" or a "time-out wait". 
+
+**Busy Wait scheme**
+
+The graph can be executed for a fixed number of iteration by ``xrtGraphRun`` API using an iteration argument. Subsequently, ``xrtGraphWait`` or ``xrtGraphEnd`` API should be used (with argument 0) to wait until graph execution is completed. 
+
+Let's review the below example
+
+- The graph is executed for 3 iterations by API ``xrtGraphRun`` with the number of iterations as an argument. 
+- The API ``xrtGraphWait(graphHandle,0)`` is used to wait till the iteration is done. 
+
+     - The API `xrtGraphWait` is used because the host code needs to execute the graph again. 
+- The Graph is executed again for 5 iteration
+- The API ``xrtGraphEnd(graphHandle,0)`` is used to wait till the iteration is done. 
+
+    - After ``xrtGraphEnd`` the same graph should not be executed. 
+
+.. code:: c
+      :number-lines: 35
+           
+           // start from reset state
+           xrtGraphReset(graphHandle);
+           
+           // run the graph for 3 iteration
+           xrtGraphRun(graphHandle, 3);
+           
+           // Wait till the graph is done 
+           xrtGraphWait(graphHandle,0);  // Use xrtGraphWait if you want to execute the graph again
+           
+           
+           xrtGraphRun(graphHandle,5);
+           xrtGraphEnd(graphHandle,0);  // Use xrtGraphEnd if you are done with the graph execution
+
+
+**Timeout wait scheme**
+
+As shown in the above example ``xrtGraphWait(graphHandle,0)`` performs a busy-wait and suspend the execution till the graph is not done. If desired a timeout version of the wait can be achieved by ``xrtGraphWaitDone`` which can be used to wait for some specified number of milliseconds, and if the graph is not done do something else in the meantime. An example is shown below
+
+.. code:: c
+      :number-lines: 35
+           
+           // start from reset state
+           xrtGraphReset(graphHandle);
+           
+           // run the graph for 100 iteration
+           xrtGraphRun(graphHandle, 100);
+           
+            while (1) {
+             auto rval  = xrtGraphWaitDone(graphHandle, 5); 
+              std::cout << "Wait for graph done returns: " << rval << std::endl;
+              if (rval == -ETIME)  {
+                   std::cout << "Timeout, reenter......" << std::endl;
+                   // Do something 
```

Comment:
This is not c-code.  But I think maybe ..code:: c means C-APIs for graph APIs?

Reply:
ok I changed to code:: c++ for code block using c++ constructs

Reply:
Why is there `xrt` in the function name since this is C++ and there are namespaces, classes, structs, scoped enums, etc. in C++?

Reply:
Also why not just `graphHandle.reset()`?